op-node/rollup/driver/sync_deriver.go

```diff
@@ -227,16 +227,27 @@ func (s *SyncDeriver) SyncStep() {
 
 	s.tryBackupUnsafeReorg()
 
-	s.Engine.TryUpdateEngine(s.Ctx)
+	// Guard: Do not send an FCU until UsafeHead is initialized.
+	// In other words, avoid sending an FCU that sets the unsafe head to genesis.
+	// This prevents side effects such as snap sync being terminated prematurely with an empty EL.
+	if s.Engine.UnsafeL2HeadInitialized() {
+		s.Engine.TryUpdateEngine(s.Ctx)
+	}
 
 	if s.Engine.IsEngineInitialELSyncing() {
-		// The pipeline cannot move forwards if doing EL sync.
-		s.Log.Debug("Rollup driver is backing off because execution engine is syncing.",
+		// The pipeline cannot move forwards if doing initial EL sync.
+		s.Log.Debug("Rollup driver is backing off because execution engine is initial EL syncing.",
 			"unsafe_head", s.Engine.UnsafeL2Head())
 		s.StepDeriver.ResetStepBackoff(s.Ctx)
 		return
 	}
 
+	if s.SyncCfg.UnsafeOnly {
```

Comment:
can we make it `&& isSequencer` ? not sure if that's available.

Reply:
Added `NeedInitialResetEngine` to the sync config and use it. Fix at https://github.com/ethereum-optimism/optimism/pull/18290/commits/79cad91a2e4e1da0d38128cc17062cdb697ae78c